tests/lib/abort-request.js

```diff
@@ -0,0 +1,54 @@
+var tape = require("tape")
+
+var abortRequest = require("../../lib/abort-request.js")
+var sendRequest = require("../../lib/send-request.js")
+
+// Polyfill responseURL property into XMLHttpRequest if it doesn't exist,
+// just for the purposes of this test
+// This polyfill is not complete; it won't show the updated location if a
+// redirection occurred, but it's fine for our purposes.
+if (!("responseURL" in XMLHttpRequest.prototype)) {
+  var nativeOpen = XMLHttpRequest.prototype.open
+  XMLHttpRequest.prototype.open = function(method, url) {
+    this.responseURL = url
+    return nativeOpen.apply(this, arguments)
+  }
+}
+
+tape("test aborting xhr request", function(t) {
+  var requestCacheBust = sendRequest.bind({
+    options: {
+      cacheBust: true,
+    },
+  })
+
+  t.test("- pending request is aborted", function(t) {
+    var r = requestCacheBust("https://httpbin.org/delay/10", {}, function() {})
```

Comment:
Is this callback ever called? Do you want to add a test that fails if it is?

Reply:
Good idea, will do that.

Reply:
Actually, that callback is never called, because if `abortRequest(r)` doesn't abort the request, the next assert (`t.equal(r.readyState, 0, "xhr readyState is '0' (ABORTED)"))` immediately fails, so I don't think we do need to add anything else in.

Reply:
Tape has a [`t.fail()`](https://github.com/substack/tape#tfailmsg) function. I think adding that will make it more apparent that the callback is not supposed to be called. What do you think?

Reply:
I did add that and then remove it again because the callback would never fire, but I take the point about making it explicit that the callback *shouldn't* fire, will add that back in.